### PR TITLE
Change sentry integration

### LINKED ISSUE #1818: Migrate to the unified Sentry Python SDK
Sentry's raven library has a deprecation notice [in their documentation](https://docs.sentry.io/clients/python/)

> Deprecated Client
> 
> This SDK has been superseded by a new unified one. The documentation here is preserved for customers using the old client. For new projects have a look at the new client documentation: [Unified Python SDK](https://docs.sentry.io/quickstart/?platform=python).

We need to update our code [anywhere the raven library](https://github.com/pydanny/cookiecutter-django/search?q=raven&unscoped_q=raven) is used. We need to replace this with an equivalent from the new library. Relevant sections:

- [Django-specific](https://docs.sentry.io/platforms/python/django/)
- [logging](https://docs.sentry.io/platforms/python/logging/) 
- [Celery](https://docs.sentry.io/platforms/python/celery/).

We need to check which parts are required for us and implement them in the template [when Sentry is selected](https://github.com/pydanny/cookiecutter-django/search?q=use_sentry&unscoped_q=use_sentry).

### PR DESCRIPTION
**Pull request by gfabricio**
fixes #1818 

A simple screenshot running a django-cookiecutter based test project with the integration using sentry-sdk: 
screenshot from 2018-10-01 18-11-08

### CODE REVIEW

File: {{cookiecutter.project_slug}}/config/settings/production.py [hunk 1]
```diff
@@ -187,69 +192,20 @@ class MediaRootS3Boto3Storage(S3Boto3Storage):
...
-    'formatters': {
-        'verbose': {
-            'format': '%(levelname)s %(asctime)s %(module)s '
-                      '%(process)d %(thread)d %(message)s'
-        },
-    },
-    'handlers': {
-        'sentry': {
-            'level': 'ERROR',
-            'class': 'raven.contrib.django.raven_compat.handlers.SentryHandler',
```
Comment (browniebroke) [inline {{cookiecutter.project_slug}}/config/settings/production.py]: Looks like we can re-use this config and just change the handler here to the [new `SentryHandler`](https://github.com/getsentry/sentry-python/blob/18cd8bf056fbee13456140cc2fc3059b8b973f3e/sentry_sdk/integrations/logging.py#L47)? I'm not too sure to understand how the logging integration works with other loggers TBH.

**Comment by webyneter** [comment]
@gfabricio mind resolving the merge conflicts?

File: {{cookiecutter.project_slug}}/requirements/production.txt [hunk 2]
```diff
@@ -8,10 +8,10 @@ psycopg2==2.7.4 --no-binary psycopg2  # https://github.com/psycopg/psycopg2
 Collectfast==0.6.2  # https://github.com/antonagestam/collectfast
 {%- endif %}
 {%- if cookiecutter.use_sentry == "y" %}
-raven==6.9.0  # https://github.com/getsentry/raven-python
+sentry-sdk==0.3.7  # https://docs.sentry.io/quickstart/?platform=python
```
[the remark this answers is not in the document]
Comment (webyneter) [reply {{cookiecutter.project_slug}}/requirements/production.txt]: Agreed.

**Comment by webyneter** [comment]
@gfabricio any news on this one?

File: {{cookiecutter.project_slug}}/config/settings/production.py [hunk 3]
```diff
@@ -187,69 +192,20 @@ class MediaRootS3Boto3Storage(S3Boto3Storage):
...
-    'dsn': SENTRY_DSN
-}
 
+sentry_logging = LoggingIntegration(
+    level=DJANGO_SENTRY_LOG_LEVEL,  # Capture info and above as breadcrumbs
+    event_level=None     # Send no events from log messages
+)
+
+sentry_sdk.init(
+    dsn="SENTRY_DSN",
```
Comment (dekoza) [inline {{cookiecutter.project_slug}}/config/settings/production.py]: I'm pretty sure this should not be quoted.

File: {{cookiecutter.project_slug}}/config/settings/production.py [hunk 4]
```diff
@@ -187,69 +192,20 @@ class MediaRootS3Boto3Storage(S3Boto3Storage):
...
-    },
-}
-
 SENTRY_CELERY_LOGLEVEL = env.int('DJANGO_SENTRY_LOG_LEVEL', logging.INFO)
-RAVEN_CONFIG = {
-    'dsn': SENTRY_DSN
-}
 
+sentry_logging = LoggingIntegration(
+    level=DJANGO_SENTRY_LOG_LEVEL,  # Capture info and above as breadcrumbs
```
Comment (dekoza) [inline {{cookiecutter.project_slug}}/config/settings/production.py]: I can't see `DJANGO_SENTRY_LOG_LEVEL` declared anywhere in this file, maybe you meant `SENTRY_CELERY_LOGLEVEL`?

File: {{cookiecutter.project_slug}}/requirements/production.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (dekoza) [reply {{cookiecutter.project_slug}}/requirements/production.txt]: Newest version is 0.6.4

File: {{cookiecutter.project_slug}}/{{cookiecutter.project_slug}}/taskapp/celery.py [hunk 5]
```diff
@@ -25,28 +26,26 @@ def ready(self):
...
-            raven_client = RavenClient(dsn=settings.RAVEN_CONFIG['dsn'])
-            raven_register_logger_signal(raven_client)
-            raven_register_signal(raven_client)
-        {%- endif %}
+    sentry_logging = LoggingIntegration(
+        level=settings.SENTRY_CELERY_LOGLEVEL,  # Capture info and above as breadcrumbs
+        event_level=None     # Send no events from log messages
+    )
+    sentry_sdk.init(dsn=settings.SENTRY_DSN, integrations=[sentry_logging, CeleryIntegration()])
+    {%- endif %}
```
[the remark this answers is not in the document]
Comment (dekoza) [reply {{cookiecutter.project_slug}}/{{cookiecutter.project_slug}}/taskapp/celery.py]: This definitely should be part of `ready` method.

**Comment by jhancia** [comment]
Is anyone using this change in production? Is it ready to use?

File: {{cookiecutter.project_slug}}/config/settings/production.py (hunk 1 above)
Comment (KrunchMuffin) [reply {{cookiecutter.project_slug}}/config/settings/production.py]: This is all no longer necessary
From the docs
> Most SDKs will now automatically collect data if available

**Comment by KrunchMuffin** [comment]
This whole section can be removed as well

https://github.com/pydanny/cookiecutter-django/blob/65d2ce07fb133f8ace1a9f943a7b54d6dec4b1ee/%7B%7Bcookiecutter.project_slug%7D%7D/config/settings/production.py#L182

**Comment by webyneter** [comment]
What about the `config.wsgi`? It's got the `application = Sentry(application)` line as well...

File: {{cookiecutter.project_slug}}/requirements/production.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (dekoza) [reply {{cookiecutter.project_slug}}/requirements/production.txt]: Actually it's 0.7.6 now - they're working fast :smile:

File: {{cookiecutter.project_slug}}/config/wsgi.py [hunk 6]
```diff
@@ -37,10 +33,6 @@
 # file. This includes Django's development server, if the WSGI_APPLICATION
 # setting points here.
 application = get_wsgi_application()
-{% if cookiecutter.use_sentry == 'y' -%}
-if os.environ.get('DJANGO_SETTINGS_MODULE') == 'config.settings.production':
-    application = Sentry(application)
```
Comment (browniebroke) [inline {{cookiecutter.project_slug}}/config/wsgi.py]: Remove the WSGI integration as they recommend using a more specific one, when possible:

> It is recommended to use an integration for your particular WSGI framework if available, as those are easier to use and capture more useful information.

File: {{cookiecutter.project_slug}}/config/settings/production.py (hunk 1 above)
Comment (elyak123) [reply {{cookiecutter.project_slug}}/config/settings/production.py]: I would be in favor on placing some sort of comment regarding on WHY logging configuration is not longer necessary, specially for people not very familiar with django's default config.

File: {{cookiecutter.project_slug}}/{{cookiecutter.project_slug}}/taskapp/celery.py [hunk 7]
```diff
@@ -27,25 +27,26 @@ def ready(self):
...
+            from sentry_sdk.integrations.logging import LoggingIntegration
+            {% if cookiecutter.use_pycharm == 'y' -%}
             # @formatter:on
-{%- endif %}
-
-            raven_client = RavenClient(dsn=settings.RAVEN_CONFIG['dsn'])
-            raven_register_logger_signal(raven_client)
-            raven_register_signal(raven_client)
+            {%- endif %}
+            sentry_logging = LoggingIntegration(
```
Comment (elyak123) [inline {{cookiecutter.project_slug}}/{{cookiecutter.project_slug}}/taskapp/celery.py]: When executed like this, it appears to create a conflict in Sentry an silence all errors from loggin at all. To me it seems to be that the initialization must be only once, if Celery and Sentry are both configured, the `init` script is executed two times. My suggestion is to remove 
```
sentry_logging = LoggingIntegration(
    level=settings.SENTRY_LOG_LEVEL,  # Capture info and above as breadcrumbs
    event_level=None,  # Send no events from log messages
)
sentry_sdk.init(dsn=settings.SENTRY_DSN, integrations=[sentry_logging, CeleryIntegration()])
```
from here and in `production.py` do something like:
```
{% if cookiecutter.use_celery == 'y' -%}
sentry_sdk.init(
    SENTRY_DSN,
    integrations=[sentry_logging, DjangoIntegration(), CeleryIntegration()]
)
{% else %}
sentry_sdk.init(
    SENTRY_DSN,
    integrations=[sentry_logging, DjangoIntegration()]
)
{%- endif %}
```

Comment (dekoza) [reply {{cookiecutter.project_slug}}/{{cookiecutter.project_slug}}/taskapp/celery.py]: @elyak123 I can't thank you enough for this comment.

Comment (browniebroke) [reply {{cookiecutter.project_slug}}/{{cookiecutter.project_slug}}/taskapp/celery.py]: @dekoza do you mean that you had the same problem and this fixed it?

Comment (dekoza) [reply {{cookiecutter.project_slug}}/{{cookiecutter.project_slug}}/taskapp/celery.py]: @browniebroke My project was not reporting errors except for errors in the settings.py. I even made a special view raising excpetion to test if sentry works. I lost several hours trying to debug this and this comment saved me from putting even more hours to this.

**Pull request merged**